src/python/WMCore/WMSpec/StdSpecs/StepChain.py

```diff
@@ -321,9 +324,17 @@ def setupNextSteps(self, task, origArgs):
                 multicore = taskConf['Multicore']
             if taskConf.get("EventStreams") is not None and taskConf['EventStreams'] >= 0:
                 eventStreams = taskConf['EventStreams']
-
             currentCmsswStepHelper.setNumberOfCores(multicore, eventStreams)
 
+            # GPU settings
+            gpuRequired = self.requiresGPU
+            gpuParams = json.loads(taskConf.get('GPUParams', 'null'))
+            if taskConf.get('RequiresGPU', None):
+                gpuRequired = taskConf['RequiresGPU']
+            if "GPUParams" not in taskConf:
```

Comment:
Thanks @amaltaro, 
But in this line here, you may still hit exactly the same set of exceptions, because the `'GPUParams'` key may be present in `taskConf` but have an empty string as a value. Maybe we rely on some previous validation for not having empty or misconfigured values for those parameters... I think the safer way would be to have all checks related to the `gpuParams` enclosed in a single `if/then/else` block. But anyway, up to you.

Reply:
This should be covered by this attribute definition:
https://github.com/dmwm/WMCore/blob/master/src/python/WMCore/WMSpec/StdSpecs/StdBase.py#L1228

which already does the data type validation as well.